### PR TITLE
{Release} Upgrade to Azure CLI 2.59.0

### PR DESCRIPTION
**Pull request by azclibot**
Updated Azure CLI version.

### CODE REVIEW

**Comment by yonzhan** [comment]
Release

File: src/azure-cli/HISTORY.rst [hunk 1]
```diff
@@ -3,6 +3,105 @@
 Release History
 ===============
 
+2.59.0
+++++++
+
+**ACR**
+
+* Fix #14768: `az acr login`: Add environment Variable for docker command (#28443)
```
Comment (zhoxing-ms) [inline src/azure-cli/HISTORY.rst:11]:
```suggestion
* Fix #14768: `az acr login`: Add environment variable for docker command (#28443)
```

File: src/azure-cli/HISTORY.rst [hunk 2]
```diff
@@ -3,6 +3,105 @@
...
+* `az aks create`: Add optional parameter `--revision` to set revision for the Azure Service Mesh addon while creating AKS cluster (#28482)
+* `az aks mesh get-upgrades`: Fix command failure with a traceback if ASM addon is disabled (#28507)
+* `az aks create/update`: Enable mooncake support for managed prometheus addon (#28504)
+* `az aks create/update`: Block Azure Managed Grafana for managed prometheus addon in air gapped cloud (#28570)
+* `az aks create`: Correct use of "comma-separated" in help (#28245)
+
+**App Config**
+
+* `az appconfig feature filter update`: GA command (#28459)
+* `az kv export`: GA parameter `--export-as-reference` (#28459)
```
Comment (zhoxing-ms) [inline src/azure-cli/HISTORY.rst:35]:
```suggestion
* `az appconfig kv export`: GA parameter `--export-as-reference` (#28459)
```

File: src/azure-cli/HISTORY.rst [hunk 3]
```diff
@@ -3,6 +3,105 @@
...
+**ACR**
+
+* Fix #14768: `az acr login`: Add environment Variable for docker command (#28443)
+
+**ACS**
+
+* `az aks create`: Add flag `--enable-app-routing` to enable app routing (#28463)
+* `az aks approuting`: Add command group to handle enable|disable|update of the approuting addon (#28463)
+* `az aks approuting zone`: Add command group to handle add|delete|update|list actions of dns zone resources associated to the approuting addon (#28463)
+* `az aks create/update`: Introduce changes for Azure container storage in ACS cli (#28251)
```
Comment (zhoxing-ms) [inline src/azure-cli/HISTORY.rst:18]:
```suggestion
* `az aks approuting`: Add command group to handle enable/disable/update of the app routing addon (#28463)
* `az aks approuting zone`: Add command group to handle add/delete/update/list actions of DNS zone resources associated to the approuting addon (#28463)
* `az aks create/update`: Introduce changes for Azure container storage in ACS CLI (#28251)
```

File: src/azure-cli/HISTORY.rst [hunk 4]
```diff
@@ -3,6 +3,105 @@
...
+* `az netappfiles check-quota-availability`: Add new command to check if a quota is available (#27951)
+* `az netappfiles query-network-sibling-set`: Add new command to describe a network sibling set (#27951)
+* `az netappfiles update-network-sibling-set`: Add new command to Update the network features of a network sibling set (#27951)
+* `az netappfiles quota-limit`: Add new command group to mange quota limits (#27951)
+* `az netappfiles volume populate-availability-zone`: Add new command to populate availability zone information for a volume (#27951)
+* `az netappfiles volume replication re-initialize`: Add new command to Re-establish a previously deleted replication between 2 volumes that have a common ad-hoc or policy-based snapshots (#27951)
+
+**Network**
+
+* `az network virtual-appliance connection`: Add update commands for NVA connection (#28461)
```
Comment (necusjz) [inline src/azure-cli/HISTORY.rst:100]:
```suggestion
* `az network virtual-appliance connection`: Add update command for NVA connection (#28461)
```

File: src/azure-cli/HISTORY.rst [hunk 5]
```diff
@@ -3,6 +3,105 @@
...
+++++++
+
+**ACR**
+
+* Fix #14768: `az acr login`: Add environment Variable for docker command (#28443)
+
+**ACS**
+
+* `az aks create`: Add flag `--enable-app-routing` to enable app routing (#28463)
+* `az aks approuting`: Add command group to handle enable|disable|update of the approuting addon (#28463)
```
Comment (yanzhudd) [inline src/azure-cli/HISTORY.rst:16]:
```suggestion
* `az aks approuting`: Add command group to handle enabling/disabling/updating of the approuting addon (#28463)
```

File: src/azure-cli/HISTORY.rst [hunk 6]
```diff
@@ -3,6 +3,105 @@
...
+* `az mysql flexible-server import create`: Add support for online migration for single to flex (#28599)
+
+**NetAppFiles**
+
+* `az netappfiles check-file-path-availability`: Add new command to check if a file path is available (#27951)
+* `az netappfiles check-name-availability`: Add new command to check if a resource name is available (#27951)
+* `az netappfiles check-quota-availability`: Add new command to check if a quota is available (#27951)
+* `az netappfiles query-network-sibling-set`: Add new command to describe a network sibling set (#27951)
+* `az netappfiles update-network-sibling-set`: Add new command to Update the network features of a network sibling set (#27951)
+* `az netappfiles quota-limit`: Add new command group to mange quota limits (#27951)
```
Comment (bebound) [inline src/azure-cli/HISTORY.rst:94]:
```suggestion
* `az netappfiles quota-limit`: Add new command group to manage quota limits (#27951)
```

File: src/azure-cli/HISTORY.rst [hunk 7]
```diff
@@ -3,6 +3,105 @@
...
+
+**ACR**
+
+* Fix #14768: `az acr login`: Add environment Variable for docker command (#28443)
+
+**ACS**
+
+* `az aks create`: Add flag `--enable-app-routing` to enable app routing (#28463)
+* `az aks approuting`: Add command group to handle enable|disable|update of the approuting addon (#28463)
+* `az aks approuting zone`: Add command group to handle add|delete|update|list actions of dns zone resources associated to the approuting addon (#28463)
```
Comment (yanzhudd) [inline src/azure-cli/HISTORY.rst:17]:
```suggestion
* `az aks approuting zone`: Add command group to handle adding/deleting/updating/listing actions of dns zone resources associated to the approuting addon (#28463)
```

File: src/azure-cli/HISTORY.rst [hunk 8]
```diff
@@ -3,6 +3,105 @@
...
+* `az mysql flexible-server advanced-threat-protection-setting update`: Update server's advanced threat protection setting using `--state` as Enabled/Disabled (#28389)
+* `az mysql flexible-server import create`: Add support for online migration for single to flex (#28599)
+
+**NetAppFiles**
+
+* `az netappfiles check-file-path-availability`: Add new command to check if a file path is available (#27951)
+* `az netappfiles check-name-availability`: Add new command to check if a resource name is available (#27951)
+* `az netappfiles check-quota-availability`: Add new command to check if a quota is available (#27951)
+* `az netappfiles query-network-sibling-set`: Add new command to describe a network sibling set (#27951)
+* `az netappfiles update-network-sibling-set`: Add new command to Update the network features of a network sibling set (#27951)
```
Comment (bebound) [inline src/azure-cli/HISTORY.rst:93]:
```suggestion
* `az netappfiles update-network-sibling-set`: Add new command to update the network features of a network sibling set (#27951)
```

**Review by wangzelin007 — APPROVED** [review]

File: src/azure-cli/HISTORY.rst [hunk 9]
```diff
@@ -3,6 +3,105 @@
...
+* `az netappfiles quota-limit`: Add new command group to mange quota limits (#27951)
+* `az netappfiles volume populate-availability-zone`: Add new command to populate availability zone information for a volume (#27951)
+* `az netappfiles volume replication re-initialize`: Add new command to Re-establish a previously deleted replication between 2 volumes that have a common ad-hoc or policy-based snapshots (#27951)
```
Comment (zhoxing-ms) [inline src/azure-cli/HISTORY.rst:96]:
```suggestion
* `az netappfiles volume replication re-initialize`: Add new command to re-establish a previously deleted replication between 2 volumes that have a common ad-hoc or policy-based snapshots (#27951)
```

**Review by jiasli — APPROVED** [review]

**Review by zhoxing-ms — APPROVED** [review]

**Review by necusjz — APPROVED** [review]

**Review by bebound — APPROVED** [review]

**Review by yanzhudd — APPROVED** [review]

**Review by evelyn-ys — APPROVED** [review]

**Review by wangzelin007 — APPROVED** [review]

**Pull request merged**